### PR TITLE
Upgrade playground dependencies

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR upgrades the playground's runtime and dev dependencies

## Test Plan

I tested the playground locally

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5830** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5830" target="_blank">Graphite</a>  👈
    * **PR #5831** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5831" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5830?utm_source=stack-comment).

File: playground/index.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (konstin) [inline playground/index.html:1]: TIL https://html.spec.whatwg.org/multipage/syntax.html#the-doctype:

> A string that is an [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) match for the string "<!DOCTYPE".

neat

File: playground/src/Editor/settings.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export function stringify(settings: Settings): string {
  */
 export async function persist(settingsSource: string, pythonSource: string) {
   const hash = lzstring.compressToEncodedURIComponent(
-    settingsSource + "$$$" + pythonSource,
+    settingsSource.replaceAll("$$$", "$$$$$$") + "$$$" + pythonSource,
```
Comment (charliermarsh) [inline playground/src/Editor/settings.ts:27]: What’s going on here?

**Review by charliermarsh — APPROVED** [review]

**Comment by MichaReiser** [comment]
@MichaReiser started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5830).

**Comment by MichaReiser** [comment]
@MichaReiser merged this pull request with [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5830).

**Pull request merged**